src/utils.cpp

```diff
@@ -309,6 +311,8 @@ std::string utils::retrieve_url(const std::string& url,
 
 		LOG(Level::ERROR, "%s: LibCURL error (%d): %s", logprefix.str(), res, errmsg);
 		buf = "";
+		curl_easy_setopt(easyhandle.ptr(), CURLOPT_ERRORBUFFER, NULL);
```

Comment:
This line was copied from another path without a comment. The code is now harder to read, and since it got two exits now, it's possible to handle something in one branch but not the other.

I think it's better to keep this call in a single place, and introduce a new intermediate variable (say, `result`) to hold the return value of the function. Then one branch of the `if` can put an error in it, and another a valid value, and at the end of the function we just return the result as-is.

A more "C++-style" solution would be to write a one-off RAII struct that will reset the buffer, but I'm not sure if it's worth the effort.

What do you think?

Reply:
https://github.com/newsboat/newsboat/pull/2609/commits/2555300525e8f4bbfdf2ea33246e065a24639642